src/Strategy/Cocoapods/PodfileLock.hs

```diff
@@ -178,25 +180,88 @@ buildGraph lockFilePath lockFile@PodLock{lockExternalSources} = do
     lockFileDir :: Path Abs Dir
     lockFileDir = parent lockFilePath
 
+    toPodSpecPath :: Text -> Text -> Text
+    toPodSpecPath podPath dependencyName = toText $ toString podPath </> (toString dependencyName <.> "podspec")
+
     replaceVendoredPods ::
       ( Has Exec sig m
       , Has Diagnostics sig m
       , Has (State PodSpecCache) sig m
+      , Has Logger sig m
       ) =>
       Dependency ->
       m Dependency
     replaceVendoredPods d@Dependency{dependencyType, dependencyName} = case dependencyType of
       PodType -> case Map.lookup dependencyName lockExternalSources of
         Just es -> case es of
-          ExternalPodSpec podSpecPath ->
-            readPodSpecAt podSpecPath <||> pure d
-          ExternalPath podPath ->
-            readPodSpecAt (toText $ toString podPath </> (toString dependencyName <.> "podspec")) <||> pure d
+          ExternalPodSpec podSpecPath -> readPodSpecAt podSpecPath <||> pure d
+          ExternalPath podPath -> readPodSpecAt (toPodSpecPath podPath dependencyName) <||> pure d
           ExternalGitType _ -> pure d
           ExternalOtherType -> pure d
-        Nothing -> pure d
+        Nothing -> do
+          -- Pod dependency can be included as part of externally sourced dependency's sub spec.
+          -- Refer to: https://guides.cocoapods.org/syntax/podspec.html#subspec
+          --
+          -- To aptly resolve such dependencies, we infer candidate parent,
+          -- and candidate sub spec from dependency's name. If and only if:
+          --  * Candidate parent spec exists within external sources
+          --  * Parent's pod spec has sub spec listed
+          --  * Parent has git source
+          --
+          -- We replace this dependency with parent's git sourced dependency.
+          let (parentSpec, candidateSubSpec) = splitOnceOn "/" dependencyName
+
+          if candidateSubSpec == mempty
```

Comment:
[optional]: use `Data.Text.null`

Reply:
> One question: is there an easy way to prevent duplicating the debug messages? If not, they're fine as-is.

I don't think so unfortunately since traverse may visit the node repeatedly! Although, we can kill the log message (since it can be confusing)

Reply:
Nope, I'd rather have noisy logs than no logs.  Let's keep it.